### ISSUE TITLE
Question about the training loss and process

### ISSUE THREAD

**Issue by LIAGM** [huanngzh/Parts2Whole#11]
Hi, 

Thanks for your great work!

When I try to use your default config setting ([configs/train-sd15.yaml](https://github.com/huanngzh/Parts2Whole/blob/main/configs/train-sd15.yaml)) to reproduce the training process, I encounter some weird issues.

Here is the training loss when I use the **single GPU** with the default config file.
![image](https://github.com/user-attachments/assets/e2ebc863-1db0-432e-813a-863c1ba017ab)

I found the training loss exploded after 8000 iterations.

Here is the validation result at the 6600 iteration:
![image](https://github.com/user-attachments/assets/d98fce28-0e93-4675-b305-0e7fc0a36fc8)

This is the validation result at the 8000 iteration:
![image](https://github.com/user-attachments/assets/6d855ff8-0ad9-4a1f-a86b-4c348efddc85)

This is the validation result at the 13000 iteration:
![image](https://github.com/user-attachments/assets/b92578e3-6893-4f62-986a-c474aa17c3f6)

The generated image from the model becomes a noise image.

Have you tested the training config or encountered such an issue?

Thanks for your time and help!

**Comment by huanngzh** [comment]
Hi, are you using the latest code? I fixed a precision conversion issue a while ago that should improve the robustness of training.

**Comment by LIAGM** [comment]
Hi,

The experiment I showed above used the codebase with commit **88ef5a0**.

I see you updated the training code a few days ago.
However, I checked the code, and it seems that you updated the CLIP encoder part to speed up the training, not for the precision conversion issue?

Do you have any other solution for the issue?
I will use the latest code (**e217655**) to train from scratch again to monitor the training.

Thanks for your time and help!

**Comment by huanngzh** [comment]
The old code directly calls `feature_extractor` to preprocess the image before encoding image. However, I found that this method sometimes results in nan loss during training, followed by loss explosion. I suspect that this method may require manually upcasting the features to fp32 and then restoring them to fp16, but I haven't studied it yet.

When using the new code (manually preprocessing the image), I have tried it on both 8 cards and 4 cards, and the training is very stable.

If you still encounter this issue, it may be related to the small batch size during training. I suggest increasing the `gradient_accumulation_steps` value in config.
